Build failure: prl-tools (Parellel Tools)

## Steps To Reproduce

Steps to reproduce the behavior:

1. Update system flake (unstable branch)
2. Ru nixswitch-rebuild
3. build prl-tools on aarch64 (M2 Mac VM with NixOS)

## Build log

```bash
warning: updating lock file '/home/hausi/.config/system/nixos-gdata/flake.lock':
• Updated input 'nur':
'github:nix-community/NUR/18153db391b7a6cf5c496984f74f78875a382ee3?narHash=sha256-XCoRiukCZ7Tvr5KhuWOGX88XblaHQTmszOyCGNfxD0Y%3D' (2024-11-21)
→ 'github:nix-community/NUR/5e67f4745975ab2e5111ca7ff6c3ae02e9a555b8?narHash=sha256-l4YncGn8Ee2v8DW3WqtKI4qHeGk7aUc3KoAPvwunC%2Bo%3D' (2024-11-21)
warning: Git tree '/home/hausi/.config/system' is dirty
warning: Git tree '/home/hausi/.config/system' is dirty
building the system configuration...
warning: Git tree '/home/hausi/.config/system' is dirty
error: builder for '/nix/store/5wn0n7c4nsdkmabkvvlqj6v2csk05yrd-prl-tools-20.1.1-55740.drv' failed with exit code 2;
last 25 log lines:
> /build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c: At top level:
> /build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c:819:5: warning: no previous prototype for 'prlfs_readpage' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmissing-prototypes-Wmissing-prototypes8;;]
>   819 | int prlfs_readpage(struct file *file, struct page *page) {
>       |     ^~~~~~~~~~~~~~
> /build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c:856:5: warning: no previous prototype for 'prlfs_writepage' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmissing-prototypes-Wmissing-prototypes8;;]
>   856 | int prlfs_writepage(struct page *page, struct writeback_control *wbc) {
>       |     ^~~~~~~~~~~~~~~
> /build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c: In function 'prlfs_writepage':
> /build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c:875:17: error: implicit declaration of function 'SetPageError'; did you mean 'SetPageDirty'? [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Werror=implicit-function-declaration8;;]
>   875 |                 SetPageError(page);
>       |                 ^~~~~~~~~~~~
>       |                 SetPageDirty
> /build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c: At top level:
> /build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c:930:27: error: initialization of 'int (*)(struct file *, struct address_space *, loff_t,  unsigned int,  unsigned int,  struct folio *, void *)' {aka 'int (*)(struct file *, struct address_space *, long long int,  unsigned int,  unsigned int,  struct folio *, void *)'} from incompatible pointer type 'int (*)(struct file *, struct address_space *, loff_t,  unsigned int,  unsigned int,  struct page *, void *)' {aka 'int (*)(struct file *, struct address_space *, long long int,  unsigned int,  unsigned int,  struct page *, void *)'} [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
>   930 |         .write_end      = prlfs_write_end,
>       |                           ^~~~~~~~~~~~~~~
> /build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.c:930:27: note: (near initialization for 'prlfs_aops.write_end')
> cc1: some warnings being treated as errors
> make[4]: *** [/nix/store/lr0ikljbkdiz846zpy11n5fdavy9dsmc-linux-6.12-dev/lib/modules/6.12.0/source/scripts/Makefile.build:229: /build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs/inode.o] Error 1
> make[3]: *** [/nix/store/lr0ikljbkdiz846zpy11n5fdavy9dsmc-linux-6.12-dev/lib/modules/6.12.0/source/Makefile:1936: /build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs] Error 2
> make[2]: *** [/nix/store/lr0ikljbkdiz846zpy11n5fdavy9dsmc-linux-6.12-dev/lib/modules/6.12.0/source/Makefile:224: __sub-make] Error 2
> make[2]: Leaving directory '/nix/store/lr0ikljbkdiz846zpy11n5fdavy9dsmc-linux-6.12-dev/lib/modules/6.12.0/build'
> make[1]: *** [Makefile:36: all] Error 2
> make[1]: Leaving directory '/build/prl-tools-build/kmods/prl_fs/SharedFolders/Guest/Linux/prl_fs'
> make: *** [Makefile.kmods:70: compile] Error 2
For full logs, run 'nix log /nix/store/5wn0n7c4nsdkmabkvvlqj6v2csk05yrd-prl-tools-20.1.1-55740.drv'.
error: 1 dependencies of derivation '/nix/store/jj5xcnpyl0isb4yyxzyh1m2ix27adxqv-hwdb.bin.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1c9wcpfyznrwwm8qap0vrba5mg73h8gq-linux-6.12-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/b2555q9d4n1356vx8ddzlb399ljyik61-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9f5daylbnzg3jjr11m580vbmds0m4aca-udev-rules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6hn1a8awsjbdb8as2n24b8g05ap5vmam-unit-prlcc.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/z0sz1sf7bnn5iwx8y2b9wsp2abr6ixin-unit-prlcp.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/c58v73493ba842n2iv99ai8y743kp2dn-unit-prldnd.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/idgbkqgkwlg7bbklavxcdh141vn449rp-unit-prlsga.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rrqimh2vvd8ih9la8pb047wngdpmmkmp-unit-prlshprint.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2qaygjy5ypg28w9fxr4nh27qdshg0gx7-unit-prlshprof.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/32bddb048d2b98wcbzi66rwm7c5xn2xb-unit-prltoolsd.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/k1wkk51a5d4ix1h6basbrpm5hqgc1p51-nixos-system-nixos-25.05.20241119.23e89b7.drv' failed to build
``

## Additional context
Problem is on a ARM64 Nixos VM on a M2 Macbook with Parallels as the Hypervisor.

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

```bash
these 7 paths will be fetched (0.69 MiB download, 10.17 MiB unpacked):
/nix/store/anmwlb2bqlxh43gk5z3jf07v5z2jfgii-binutils-wrapper-2.43.1
/nix/store/nl2xzjmmz76kbicw2h7wkkxk8bx8n1dq-file-5.45
/nix/store/g45nm13bwh1jxypl33wxi016yjzbidkp-gcc-wrapper-13.3.0
/nix/store/l6mcqhhm85f13qzh9nj7jlbr3k83iyqc-gnu-config-2024-01-01
/nix/store/sjxx419av3k62z1a0fynqfywca65qlmx-gnumake-4.4.1
/nix/store/jvq7h4nv8fvi0l2s7a9b6dm00fcy8sv6-stdenv-linux
/nix/store/j8vpq5m2z9mnwy10lhsg210s0x5rvh4y-update-autotools-gnu-config-scripts-hook
copying path '/nix/store/l6mcqhhm85f13qzh9nj7jlbr3k83iyqc-gnu-config-2024-01-01' from 'https://cache.nixos.org'...
copying path '/nix/store/nl2xzjmmz76kbicw2h7wkkxk8bx8n1dq-file-5.45' from 'https://cache.nixos.org'...
copying path '/nix/store/sjxx419av3k62z1a0fynqfywca65qlmx-gnumake-4.4.1' from 'https://cache.nixos.org'...
copying path '/nix/store/anmwlb2bqlxh43gk5z3jf07v5z2jfgii-binutils-wrapper-2.43.1' from 'https://cache.nixos.org'...
copying path '/nix/store/j8vpq5m2z9mnwy10lhsg210s0x5rvh4y-update-autotools-gnu-config-scripts-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/g45nm13bwh1jxypl33wxi016yjzbidkp-gcc-wrapper-13.3.0' from 'https://cache.nixos.org'...
copying path '/nix/store/jvq7h4nv8fvi0l2s7a9b6dm00fcy8sv6-stdenv-linux' from 'https://cache.nixos.org'...
- system: `"aarch64-linux"`
- host os: `Linux 6.11.7, NixOS, 24.11 (Vicuna), 24.11.20241109.76612b1`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.24.10`
- nixpkgs: `/nix/store/0sxsfzswhjck7f4sbsvznz285s4i7y64-source`

```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@codgician 
@Artturin 
---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/prl-tools/default.nix
+++ b/pkgs/os-specific/linux/prl-tools/default.nix
@@ -41,13 +41,13 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "prl-tools";
-  version = "20.1.3-55743";
+  version = "20.2.0-55872";
 
   # We download the full distribution to extract prl-tools-lin.iso from
   # => ${dmg}/Parallels\ Desktop.app/Contents/Resources/Tools/prl-tools-lin.iso
   src = fetchurl {
     url = "https://download.parallels.com/desktop/v${lib.versions.major finalAttrs.version}/${finalAttrs.version}/ParallelsDesktop-${finalAttrs.version}.dmg";
-    hash = "sha256-5lbTTQucop/jnsVudoqTO9bESR5tdn8NFu9Nm2WphU4=";
+    hash = "sha256-oOilbF5MzZxZXNVQYAp/JxyMVdM0oltG8pGfzzsQ1kY=";
   };
 
   hardeningDisable = [
@@ -115,7 +115,6 @@ stdenv.mkDerivation (finalAttrs: {
     ( # kernel modules
       cd kmods
       mkdir -p $out/lib/modules/${kernelVersion}/extra
-      cp prl_fs/SharedFolders/Guest/Linux/prl_fs/prl_fs.ko $out/lib/modules/${kernelVersion}/extra
       cp prl_fs_freeze/Snapshot/Guest/Linux/prl_freeze/prl_fs_freeze.ko $out/lib/modules/${kernelVersion}/extra
       cp prl_tg/Toolgate/Guest/Linux/prl_tg/prl_tg.ko $out/lib/modules/${kernelVersion}/extra
       ${lib.optionalString stdenv.hostPlatform.isAarch64 "cp prl_notifier/Installation/lnx/prl_notifier/prl_notifier.ko $out/lib/modules/${kernelVersion}/extra"}
@@ -168,9 +167,6 @@ stdenv.mkDerivation (finalAttrs: {
         cp $i $out/lib
         ln -s $out/$i $out/''${i%.0.0}
       done
-
-      mkdir -p $out/share/man/man8
-      install -Dm644 ../mount.prl_fs.8 $out/share/man/man8
 
       substituteInPlace ../99prltoolsd-hibernate \
         --replace "/bin/bash" "${bash}/bin/bash"

--- a/pkgs/os-specific/linux/prl-tools/update.sh
+++ b/pkgs/os-specific/linux/prl-tools/update.sh
@@ -37,5 +37,4 @@ sha256="$(nix store prefetch-file $dmg_url --json | jq -r '.hash')"
 sed -i -e "s,version = \"$old_version\",version = \"$version\"," \
     -e "s,hash = \"sha256-.*\",hash = \"$sha256\"," "$path"
 
-git commit -qm "linuxPackages_latest.prl-tools: $old_version -> $version" "$path"
 echo "Updated linuxPackages_latest.prl-tools $old_version -> $version"

